src/components/shared/MarkerContextMenu.js

```diff
@@ -238,6 +238,22 @@ class MarkerContextMenuImpl extends PureComponent<Props> {
         <MenuItem onClick={this.setEndRange}>
           Set selection end time here
         </MenuItem>
+        <SubMenu title="Set selection start ">
+          <MenuItem onClick={this.setStartRange}>
+            From the start of this marker
+          </MenuItem>
+          <MenuItem onClick={this.setEndRange}>
+            From the end of this marker
+          </MenuItem>
+        </SubMenu>
+        <SubMenu title="Set selection end ">
+          <MenuItem onClick={this.setStartRange}>
+            From the start of this marker
+          </MenuItem>
+          <MenuItem onClick={this.setEndRange}>
+            From the end of this marker
+          </MenuItem>
+        </SubMenu>
```

Comment:
You need to add new methods and use them here. Indeed `setStartRange` sets the start of the range from marker.start, and `startEndRange` sets the end of the range from marker.end. But here we want also to set the start of the range from marker end, and set the end of the range from marker start.

So I suggest to:
1. rename the existing methods `setStartRange` => `setStartRangeFromMarkerStart` and `setEndRange` => `setEndRangeFromMarkerEnd`. You'll noticed that `setEndRangeFromMarkerEnd` can take the `marker.start` if `marker.end` is absent, I think that's for dot markers too so we can probably remove this (but we can also keep it, that does no harm really).
2. create new methods for the 2 other cases, and use them all here

Also the 2 first items should be replaced by the new submenu-based items -- except when it's a "zero duration marker".